### DISCUSSION TITLE
How to use vscode-json-languageserver to format json?

### DISCUSSION THREAD

**Discussion by limaceous-bushwhacker** [LazyVim/LazyVim#2906] [Q&A]
I found this snippet but it isn't working. Am I missing something?
```
	{
		"neovim/nvim-lspconfig",
		opts = {
			servers = {
				jsonls = {
					cmd = { "vscode-json-languageserver", "--stdio" },
					filetypes = { "json" },
					settings = {
						format = {
							enable = true,
						},
						validate = { enable = true },
					},
				},
			},
		},
	},
```

**Comment by 0x2E** [comment]
Have you installed the LSP in mason? Try this `:MasonInstall json-lsp`.

BTW, You can enable JSON by LazyVim Extras: https://www.lazyvim.org/extras/lang/json

**Reply by limaceous-bushwhacker** [reply]
Hey, I've enabled JSON support through Extras but I'm not using Mason to install language services (I'm using Nix) and in Nix packages the server is named as `vscode-json-languageserver`. That throws an error (_Spawning language server failed_), so I had to resort to manually specifying the `cmd` in `jsonls`. Does `vscode-json-languageserver` also support formatting? Or do I have to install a formatter separately? FWIW I can use jq just fine.

**Reply by 0x2E** [reply]
> Does vscode-json-languageserver also support formatting?

Yes.

---

Maybe you have to configure `conform.nvim` manually ([doc](https://www.lazyvim.org/plugins/formatting)): 

- insert `formatters_by_ft` like this: https://github.com/LazyVim/LazyVim/blob/97480dc5d2dbb717b45a351e0b04835f138a9094/lua/lazyvim/plugins/extras/formatting/prettier.lua#L31
- set the path to jsonls in `formatters` like this: https://github.com/stevearc/conform.nvim#customizing-formatters

**Reply by 0x2E** [reply]
If it is still broken, `:ConformInfo` may be helpful.

My `:ConformInfo`:

```text
...
Formatters for this buffer:
LSP: jsonls
...
```

**Reply by limaceous-bushwhacker** [reply]
Yeah, looks like Conform can't find `vscode-json-languageserver`. It says _config not found_. I've managed to get `fixjson` to work as a formatter. Guess it'll have to do for now.

This is my current config:
```
return {
	{
		"neovim/nvim-lspconfig",
		opts = {
			servers = {
				jsonls = {
					cmd = { "vscode-json-languageserver", "--stdio" },
				},
			},
		},
	},
	{
		"stevearc/conform.nvim",
		opts = {
			formatters_by_ft = {
				json = { "fixjson" },
			},
		},
	},
	{ "williamboman/mason-lspconfig.nvim", enabled = false },
	{ "williamboman/mason.nvim", enabled = false },
}
```
